extensions/resource-deployment/src/ui/toolsAndEulaSettingsPage.ts

```diff
@@ -216,6 +223,7 @@ export class ToolsAndEulaPage extends ResourceTypePage {
 						}));
 
 						this._optionDropDownMap.set(option.name, optionSelectBox);
+						this.wizard.provider = this.getCurrentProvider();
```

Comment:
this line of change doesn't seem to be related to the option level filtering. Are you fixing another issue here?

Reply:
Sort of. This is something that is called whenever you change the selected option to update the provider. Without it it just defaults to the first one - which wasn't necessarily correct when you filter out items. So it was never really an issue before because we only ever selected the first one anyways and then it would be updated correctly when another option was selected.